src/engine/specs/CoreSpecs.ts

```diff
@@ -107,12 +107,27 @@ export interface ICustomMeshSpec {
   scale?: Vector3d;
 }
 
+export interface IRectangleZoneSpec {
+  xLength: number;
+  zLength: number;
+}
+
+export interface IEllipseZoneSpec {
+  xRadius: number;
+  zRadius: number;
+}
+
+export interface IPolygonZoneSpec {
+  points: Vector2d[];
+}
+
 // Zone
 export interface IZoneSpec extends IBaseSimObjectSpec {
   type: "zone";
   zoneId: string;
-  xLength: number;
-  zLength: number;
+  rectangleZone?: IRectangleZoneSpec;
```

Comment:
Could have a `shape` field here that is of type `ZoneShape`

Then, have:

```typescript
interface IBaseZoneShape {
  type: string
}

interface IRectangleZoneShape extends IBaseZoneShape {
  type: "rectangle";
  xLength: number;
  zLength: number;
}

// Do the same for Ellipse and Polygon

type ZoneShape = IRectangleZoneShape | IEllipseZoneShape | IPolygonZoneShape;
```

This forces a valid shape to be provided, and in the constructor, you can then just look at "type" to figure out the shape that it is